packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -355,43 +360,178 @@ export async function removeProjectFromExperiments(
 }
 
 export async function getExperimentByIdea(
+  organization: string,
   idea: IdeaDocument
 ): Promise<ExperimentInterface | null> {
   const experiment = await ExperimentModel.findOne({
-    organization: idea.organization,
+    organization,
     ideaSource: idea.id,
   });
 
   return experiment ? toInterface(experiment) : null;
 }
 
-export async function getExperimentsByQuery(
-  query: FilterQuery<ExperimentDocument>,
-  projections: { [key: string]: boolean },
-  limit?: number,
-  sortByField?: string,
-  sortAscending?: boolean
-): Promise<ExperimentInterface[]> {
-  // If a sortByField is specified, add sort method, otherwise run query without
-  if (sortByField) {
-    const experiments = await ExperimentModel.find(query, projections)
-      .limit(limit || 0)
-      .sort({
-        sortByField: sortAscending ? 1 : -1,
-      });
-
-    return experiments.map((m) => toInterface(m));
-  } else {
-    const experiments = await ExperimentModel.find(query, projections).limit(
-      limit || 0
-    );
-    return experiments.map((m) => toInterface(m));
-  }
+export async function getExperimentsToUpdate(
+  ids: string[]
+): Promise<Pick<ExperimentInterface, "id" | "organization">[]> {
+  const experiments = await ExperimentModel.find(
+    {
+      datasource: {
+        $exists: true,
+        $ne: "",
+      },
+      status: "running",
+      autoSnapshots: true,
+      nextSnapshotAttempt: {
+        $exists: true,
+        $lte: new Date(),
+      },
+      id: {
+        $nin: ids,
+      },
+    },
+    {
+      id: true,
+      organization: true,
+    }
+  )
+    .limit(100)
+    .sort({ nextSnapShotAttempt: 1 });
+  return experiments.map((m) => toInterface(m));
+}
+
+export async function getExperimentsToUpdateLegacy(
+  latestDate: Date
+): Promise<Pick<ExperimentInterface, "id" | "organization">[]> {
```

Comment:
@jdorn For all of these new methods that replaced the general `getExperimentsByQuery` method, I opted to define the return type in-line. It does get a bit long with the `Pick` statement - let me know if you'd prefer I define these above and reference here.